.github/workflows/tests.yaml

```diff
@@ -11,104 +12,73 @@ on:
       - master
 
 jobs:
-  run_tox:
-    name: tox -e ${{ matrix.toxenv }} (${{matrix.python-version}} on ${{ matrix.os }})
-    runs-on: ${{ matrix.os }}
+  run-tox:
+    name: tox -e ${{ matrix.tox.env }}${{ matrix.tox.platform-label || '' }}
+    runs-on: ${{ matrix.tox.os || matrix.default-os }}
     strategy:
       fail-fast: false
       matrix:
+        default-os:
+          - "ubuntu-latest"
         python-version:
           - "3.12"
-        python-dev-version:
-          - ""
-        os:
-          - "ubuntu-latest"
-        toxenv:
-          - "pep8"
-          - "pep8-docstrings"
-          - "pep8-examples"
-          - "ruff"
-          - "mypy"
-          - "mypy_tests"
-          - "py312"
-          - "py312_sans_msgpack"
-          - "py312_cython"
-          - "docs"
-          - "towncrier"
-          - "look"
-          - "asgilook"
-          - "ws_tutorial"
-          # NOTE(vytas): Temporarily disabled while we are releasing a
-          #   new version of python-mimeparse, and haven't decided yet whether
-          #   to keep vendoring or reimplement.
-          # - "check_vendored"
-          - "twine_check"
-          - "daphne"
-          - "hypercorn"
-          - "e2e_chrome"
-          - "e2e_firefox"
-          - "no_optional_packages"
-          # TODO(kgriffs): Re-enable once hug has a chance to address
-          # breaking changes in Falcon 3.0
-          # - "hug"
-        include:
-          - python-version: pypy3.9
-            os: ubuntu-latest
-            toxenv: pypy3
-          - python-version: "3.8"
-            os: ubuntu-latest
-            toxenv: py38
-          - python-version: "3.9"
-            os: ubuntu-latest
-            toxenv: py39
-          - python-version: "3.9"
-            os: ubuntu-latest
-            toxenv: py39_cython
-          - python-version: "3.10"
-            os: ubuntu-latest
-            toxenv: py310
-          - python-version: "3.10"
-            os: ubuntu-latest
-            toxenv: py310_cython
-          - python-version: "3.11"
-            os: ubuntu-latest
-            toxenv: py311
-          - python-version: "3.11"
-            os: ubuntu-latest
-            toxenv: py311_cython
-          - python-version: "3.12"
-            os: ubuntu-latest
-            toxenv: py312
-          - python-version: "3.12"
-            os: ubuntu-latest
-            toxenv: py312_cython
-          - python-version: "3.12"
+        tox:
```

Comment:
neat. I did not know you could pass objects here

Reply:
Already using this in CIBW and elsewhere :)